### PR TITLE
feat: hide component id

### PR DESCRIPTION
**Pull request by mikechesterwang**
component id is not user-friendly, since it is not the single source of truth of the resource spec, we can hide the details and let the code do the work.

### CODE REVIEW

File: examples/resources/cluster/resource.tf [hunk 1]
```diff
@@ -1,27 +1,42 @@
...
-  resourcev1 = {
+resource "risingwavecloud_cluster" "my_cluster" {
+  region  = "us-east-1"
+  name    = "my_cluster"
+  version = "v1.8.0"
+  spec = {
     compute = {
-      type    = "p-1c2g"
-      replica = 1
+      default_node_group = {
```
Comment (arkbriar) [inline examples/resources/cluster/resource.tf:7]: Question here: can there be more node groups?

**Review by arkbriar — APPROVED** [review]
LGTM!

File: examples/resources/cluster/resource.tf [hunk 2]
```diff
@@ -1,27 +1,42 @@
...
+        replica = 1
+      }
     }
     compactor = {
-      type    = "p-2c4g"
-      replica = 3
-    },
+      default_node_group = {
+        cpu     = "1"
+        memory  = "4 GB"
```
Comment (arkbriar) [inline examples/resources/cluster/resource.tf:16]: What If the cpu/memory values are invalid?

File: examples/resources/cluster/resource.tf (hunk 1 above)
Comment (mikechesterwang) [reply examples/resources/cluster/resource.tf:7]: yes, extra node group will be supported in the later version as an attribute or a resource like `cluster_node_group`.  We can discuss the details later.

File: examples/resources/cluster/resource.tf (hunk 2 above)
Comment (mikechesterwang) [reply examples/resources/cluster/resource.tf:16]: The error message will contain the available settings: https://github.com/risingwavelabs/terraform-provider-risingwavecloud/pull/20/files#diff-39b6750b80d05e2b19ce3ccfbc3c89b47161d14d65e76a22605a1cb434fb0d7cR287

**Pull request merged**